examples/fabric-admin/device_manager/CommissionerControl.cpp

```diff
@@ -0,0 +1,142 @@
+#include "CommissionerControl.h"
+#include <device_manager/DeviceManager.h>
+
+using namespace ::chip;
+
+void CommissionerControl::Init(Controller::DeviceCommissioner & commissioner, NodeId nodeId, EndpointId endpointId)
+{
+    ChipLogProgress(NotSpecified, "Initilize CommissionerControl");
+    mCommissioner  = &commissioner;
+    mDestinationId = nodeId;
+    mEndpointId    = endpointId;
+}
+
+CHIP_ERROR CommissionerControl::RequestCommissioningApproval(uint64_t requestId, uint16_t vendorId, uint16_t productId,
+                                                             Optional<CharSpan> label)
+{
+    VerifyOrReturnError(mCommissioner != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    ChipLogProgress(NotSpecified, "Sending RequestCommissioningApproval to node " ChipLogFormatX64,
+                    ChipLogValueX64(mDestinationId));
+
+    mRequestCommissioningApproval.requestID = requestId;
+    mRequestCommissioningApproval.vendorID  = static_cast<VendorId>(vendorId);
+    mRequestCommissioningApproval.productID = productId;
+    mRequestCommissioningApproval.label     = label;
```

Comment:
What is the guarantee that the lifetime of the span, if provided will still be around when `OnDeviceConnectedFn` is called?

Reply:
Acked,  Instead of just storing the span reference, make a deep copy of the data into a member variable. This ensures that the data remains valid for the lifetime of the CommissionerControl instance.